### PR TITLE
Debug deploy function for cozy stack.

### PR DESCRIPTION
**Pull request by poupotte**

### CODE REVIEW

**Comment by jsilvestre** [comment]
@frankrousseau let me know when you have reviewed/merged/published. Thanks!

**Comment by jsilvestre** [comment]
New commits look good to me.

File: src/application.coffee [hunk 1]
```diff
@@ -181,4 +211,24 @@ class exports.ApplicationManager
                 process.kill(pid)
             catch
                 log.info 'No process.'
-        callback()
\ No newline at end of file
+        callback()
+
+    checkVersions: (appData, callback) =>
```
Comment (frankrousseau) [inline src/application.coffee]: Can you add a comment to describe this function?

File: src/application.coffee [hunk 2]
```diff
@@ -121,10 +118,43 @@ class exports.ApplicationManager
                 else
                     callback null, true
 
+    stackVersions: (callback) ->
```
Comment (frankrousseau) [inline src/application.coffee]: Can you add a comment to describe this function?

File: src/application.coffee [hunk 3]
```diff
@@ -121,10 +118,43 @@ class exports.ApplicationManager
...
+            , callback
+
     addInDatabase: (manifest, callback) ->
         dsClient = new Client 'http://localhost:9101'
         dsClient.post 'data/', manifest, (err, res, body) ->
-            callback err
+            return callback err if err?
+            if manifest.iconPath?
+                path = "data/#{body._id}/attachments/"
+                data = 'name': "icon.#{manifest.iconType}"
```
Comment (frankrousseau) [inline src/application.coffee]: `name` doesn't require `'`.

File: src/project.coffee [hunk 4]
```diff
@@ -74,4 +74,16 @@ class exports.ProjectManager
             manifest.password = 'test'
             manifest.docType = "Application"
             manifest.port = port
+            basePath = path.join process.cwd(), "client", "app", "assets", "icons"
+            svgPath = path.join basePath, "main_icon.svg"
+            pngPath = path.join basePath, "main_icon.png"
+            if manifest['icon-path'] and fs.existsSync manifest['icon-path']
```
Comment (frankrousseau) [inline src/project.coffee]: A blank line above the if would makes things clearer.

File: src/project.coffee [hunk 5]
```diff
@@ -74,4 +74,16 @@ class exports.ProjectManager
...
+                manfest.iconPath = manifest['icon-path']
+                manifest.iconType = 'png'
+            else if fs.existsSync(svgPath)
```
Comment (frankrousseau) [inline src/project.coffee]: A blank line above the if would makes things clearer.

File: src/project.coffee [hunk 6]
```diff
@@ -74,4 +74,16 @@ class exports.ProjectManager
...
+                manifest.iconType = 'svg'
+                manifest.iconPath = svgPath
+            else if fs.existsSync(pngPath)
```
Comment (frankrousseau) [inline src/project.coffee]: A blank line above the if would makes things clearer.

**Comment by frankrousseau** [comment]
Thank you, published as 1.0.17

**Pull request merged**